src/hooks/use-fetch.ts

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useReducer, useRef } from "react";
+
+interface State<T> {
```

Comment:
This probably seems like something worth adding a test for

Reply:
I\'d prefer to do this mono repo side rathter than here as the implementation here is dumber than the monorepo one and once we get this project into the mono repo we can share this from react helpers